plugins/guests/windows/cap/change_host_name.rb

```diff
@@ -27,13 +30,26 @@ def self.change_host_name_and_wait(machine, name, sleep_timeout)
             error_class: Errors::RenameComputerFailed,
             error_key: :rename_computer_failed)
 
-          # Don't continue until the machine has shutdown and rebooted
-	  if machine.guest.capability?(:wait_for_reboot)
-            machine.guest.capability(:wait_for_reboot)
-          else
-            # use graceful_halt_timeout only if guest cannot wait for reboot
-            sleep(sleep_timeout)
-	  end
+          timeout = 0
+          while true
+            begin
+              # Don't continue until the machine has shutdown and rebooted
+              if machine.guest.capability?(:wait_for_reboot)
+                machine.guest.capability(:wait_for_reboot)
+                break
+              else
+                @logger.debug("No wait_for_reboot capability, sleeping for #{sleep_timeout} instead...")
+                # use graceful_halt_timeout only if guest cannot wait for reboot
+                sleep(sleep_timeout)
+                break
+              end
+            rescue Vagrant::Errors::MachineGuestNotReady => e
+              raise if timeout == 24
```

Comment:
Instead of having a `timeout` variable that is incremented and checked it might be better to have a known amount of time we will attempt retries and store that in a constant (like `MAX_REBOOT_DURATION`). Then we can use that value and decrement it on each retry.

```ruby
MAX_REBOOT_DURATION = 120

wait_remaining = MAX_REBOOT_DURATION
begin
  ...
rescue Vagrant::Errors::MachineGuestNotReady
  raise if wait_remaining < 0
  sleep(5)
  wait_remaining -= 5
  retry
end
```

Having the time defined in a constant will help to make it clearer in the future how much time is allowed if it needs to be adjusted for some reason.